openpype/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -25,27 +25,48 @@ class SubstanceLoadProjectMesh(load.LoaderPlugin):
     icon = "code-fork"
     color = "orange"
 
-    options = [
-        qargparse.Boolean(
-            "preserve_strokes",
-            default=True,
-            help="Preserve strokes positions on mesh.\n"
-                 "(only relevant when loading into existing project)"
-        ),
-        qargparse.Boolean(
-            "import_cameras",
-            default=True,
-            help="Import cameras from the mesh file."
-        )
-    ]
-
-    def load(self, context, name, namespace, data):
+    @classmethod
+    def get_options(cls, contexts):
+        project_workflow_option = {
+            substance_painter.project.ProjectWorkflow.Default: "default",
+            substance_painter.project.ProjectWorkflow.UVTile: "uvTile",
+            substance_painter.project.ProjectWorkflow.TextureSetPerUVTile: "textureSetPerUVTile"     # noqa
+        }
+        return [
+            BoolDef("preserve_strokes",
+                    default=True,
+                    label="Preserve Strokes",
+                    tooltip=("Preserve strokes positions on mesh.\n"
+                             "(only relevant when loading into "
+                             "existing project)")),
+            BoolDef("import_cameras",
+                    default=True,
+                    label="Import Cameras",
+                    tooltip="Import cameras from the mesh file."
+            ),
+            EnumDef("texture_resolution",
+                    items=[128, 256, 512, 1024, 2048, 4096],
+                    default=1024,
+                    label="Texture Resolution",
+                    tooltip="Set texture resolution when creating new project"),
+            EnumDef("project_uv_workflow",
+                    items=project_workflow_option,
+                    default="default",
+                    label="UV Workflow",
+                    tooltip="Set UV workflow when creating new project")
+        ]
+
+    def load(self, context, name, namespace, options=None):
 
         # Get user inputs
-        import_cameras = data.get("import_cameras", True)
-        preserve_strokes = data.get("preserve_strokes", True)
+        import_cameras = options.get("import_cameras", True)
+        preserve_strokes = options.get("preserve_strokes", True)
+        texture_resolution = options.get("texture_resolution", 1024)
+        uv_workflow = options.get("project_uv_workflow", "default")
```

Comment:
Shouldn't this fallback to `substance_painter.project.ProjectWorkflow.Default` instead of the string "default"?

```suggestion
        uv_workflow = options.get("project_uv_workflow", "default")
```

Reply:
Yes,  it's still a little bit confused of how the dict items to be used as enum definition. The uv_workflow argument only `recongnize substance_painter.project.ProjectWorkflow.Default` instead of the string `default`

Reply:
This is not resolved.

When `options` dialog is not used the defualts are NOT passed. As such, this should change.

```suggestion
        uv_workflow = options.get("project_uv_workflow", substance_painter.project.ProjectWorkflow.Default)
```

But then considering the line length ;)

Reply:
> This is not resolved.
> 
> When `options` dialog is not used the defualts are NOT passed. As such, this should change.
> 
> But then considering the line length ;)
> 
> This should also fix the issue mentioned [here](https://github.com/ynput/OpenPype/pull/6262#pullrequestreview-2000970524)

I actually make a function at the commit [9e9c904](https://github.com/ynput/OpenPype/pull/6262/commits/9e9c90404d5be4b821a1e26fb80623b5c2795ce0) regarding to the change, but not sure if it is the best way to implement that